### ISSUE TITLE
Semantic padding parameter for tch::nn::ConvConfigND

### ISSUE THREAD

**Issue by jerry73204** [LaurentMazare/tch-rs#45]
I raised this question from my experience in Keras/TensorFlow. Keras allows users to specify 'same'/'valid' padding besides integers. That's a great convenience. I see that PyTorch does not provide this feature, and thus it's asked on their [forum](https://discuss.pytorch.org/t/how-to-keep-the-shape-of-input-and-output-same-when-dilation-conv/14338).

I would propose a solution that may get the best of two. We could provide a method `set_padding()` on `tch::nn::ConvConfigND`. Following [Rust's convention](https://doc.rust-lang.org/std/fs/struct.OpenOptions.html), we could write:

```rust
let config = ConvConfig { /* ... */ }.set_padding("same");
```

We can let `set_padding()` to accept an enum over str or int. Likely `bias`, `strides` and other fields may apply for this change.

**Comment by LaurentMazare** [comment]
Thanks for suggesting this.
There is a longer thread about why PyTorch does not support this on github: https://github.com/pytorch/pytorch/issues/3867 After reading I feel a bit that we should do the same and not support 'same' explicitly but rather let people write their own helper function (or maybe provide one), my understanding of the thread being that 'same' padding is dependent on the input size and handling it in the same way as tensorflow results in quite some hidden operations under the hood.

As per setting bias/strides/numerical values for padding, indeed a builder pattern could be a good idea. I gave a try to `derive_builder` a while back but found it a bit too verbose.

**Comment by jerry73204** [comment]
Ok, we can keep `padding` unchanged. I could try to implement builder pattern if I have time.

**Comment by LaurentMazare** [comment]
Closing this for now, there is an helper function for `same` padding in the efficientnet implementation:
https://github.com/LaurentMazare/tch-rs/blob/master/src/vision/efficientnet.rs#L69-L88
If other models start using it, we could move the code to some common place.
